### PR TITLE
Enable oracle tests

### PR DESCRIPTION
**Pull request by koppor**
Enable test on Oracle XE. Fixes https://github.com/koppor/jabref/issues/151.

Oracle XE encrypted available at files.jabref.org. We have the password stored at koppor's KeePass and in the environment variable during the build. Thus, it can be decrypted during the test, but (hopefully) not by other persons.

- Remove obsolete ";" at SQL statement (ignored by non-Oracle-DB-systems)
- create prepared statement
- reformat code

### CODE REVIEW

**Comment by koppor** [comment]
Login into a local docker image works (after creating the jabref user)

```bash
$ docker exec -it  oracle-xe bash -c "source /home/oracle/.bashrc; /bin/bash -c '$ORACLE_HOME/bin/sqlplus jabref/jabref@localhost/XEPDB1'"
```

Then both following statements work:

```SQL
SELECT ENTRY.SHARED_ID, ENTRY.TYPE, ENTRY.VERSION, F.ENTRY_SHARED_ID, F.NAME, F.VALUE FROM ENTRY inner join FIELD F on ENTRY.SHARED_ID = F.ENTRY_SHARED_ID order by SHARED_ID;
```

```SQL
SELECT "ENTRY"."SHARED_ID", "ENTRY"."TYPE", "ENTRY"."VERSION", F."ENTRY_SHARED_ID", F."NAME", F."VALUE" FROM "ENTRY" inner join "FIELD" F on "ENTRY"."SHARED_ID" = F."ENTRY_SHARED_ID" order by "SHARED_ID";
```

But not when using the oracleJDBC driver: 

```
23:14:30.001 [main] ERROR org.jabref.logic.shared.DBMSProcessor - Executed >SELECT ENTRY.SHARED_ID, ENTRY.TYPE, ENTRY.VERSION, F.ENTRY_SHARED_ID, F.NAME, F.VALUE FROM ENTRY inner join FIELD F on ENTRY.SHARED_ID = F.ENTRY_SHARED_ID where SHARED_ID in (1) order by SHARED_ID;< -  
23:14:30.001 [main] ERROR org.jabref.logic.shared.DBMSProcessor - SQL Error - java.sql.SQLSyntaxErrorException: ORA-00933: SQL-Befehl wurde nicht korrekt beendet

	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550) 
```

The simplest test case probably is: `org.jabref.logic.shared.DBMSProcessorTest#testGetSharedEntry`

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -128,6 +128,37 @@ jobs:
...
+          unzip master.zip
+          cd docker-oracle-xe-master
+          wget -qN http://ccrypt.sourceforge.net/download/1.11/ccrypt-1.11.linux-x86_64.tar.gz
+          wget -qN -O files/oracle-database-xe-18c-1.0-1.x86_64.rpm.cpt https://files.jabref.org/thing.cpt
+          pwd
+          tar xzf ccrypt-1.11.linux-x86_64.tar.gz
+          ccrypt-1.11.linux-x86_64/ccrypt -d -E CCRYPT files/oracle-database-xe-18c-1.0-1.x86_64.rpm.cpt
+          docker build -t oracle-xe:18c .
+          docker run -d -p 32118:1521 -p 35518:5500 --name=oracle-xe oracle-xe:18c
+          docker exec -i oracle-xe bash -c "source /home/oracle/.bashrc; /bin/bash -c '$ORACLE_HOME/bin/sqlplus sys/Oracle18@localhost/XEPDB1 as sysdba'" < buildres/setup-test-oracle.sql
```
Comment (ungerts) [inline .github/workflows/tests.yml]: Maybe use environment variable _GITHUB_WORKSPACE_.

$GITHUB_WORKSPACE/buildres/setup-test-oracle.sql

File: src/main/java/org/jabref/logic/shared/DBMSProcessor.java [hunk 2]
```diff
@@ -489,12 +497,14 @@ public void removeEntries(List<BibEntry> bibEntries) {
...
                 }
 
+                // In all cases, we set the field value of the newly created BibEntry object
                 String value = selectEntryResultSet.getString("VALUE");
                 if (value != null) {
                     bibEntry.setField(FieldFactory.parseField(selectEntryResultSet.getString("NAME")), value, EntriesEventSource.SHARED);
                 }
             }
         } catch (SQLException e) {
+            LOGGER.error("Executed >{}<", query.toString());
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/shared/DBMSProcessor.java]: Shouldn't this be a log / debug level?

File: src/main/java/org/jabref/logic/shared/DBMSProcessor.java [hunk 3]
```diff
@@ -466,20 +463,31 @@ public void removeEntries(List<BibEntry> bibEntries) {
              .append(" = F.").append(escape("ENTRY_SHARED_ID"));
 
         if (!sharedIDs.isEmpty()) {
-            String idListAsString = sharedIDs.stream().map(String::valueOf).collect(Collectors.joining(", "));
             query.append(" where ")
                  .append(escape("SHARED_ID")).append(" in (")
-                 .append(idListAsString)
-                 .append(")");
+                 .append("?, ".repeat(sharedIDs.size() - 1))
```
[the remark this answers is not in the document]
Comment (koppor) [reply src/main/java/org/jabref/logic/shared/DBMSProcessor.java]: Made id consistent with other parts of DBMSProcessor (saw it in the last pull requests going through. I especially reused the idea of https://github.com/JabRef/jabref/pull/5659#discussion_r349561890).

File: .github/workflows/tests-oracle.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          restore-keys: |
+            ${{ runner.OS }}-gradle-${{ env.cache-name }}-
+            ${{ runner.OS }}-gradle-
+            ${{ runner.OS }}-
+      - uses: actions/cache@v1
+        name: Restore gradle wrapper
+        with:
+          path: ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
+      - name: Cache Oracle XE downloads
```
Comment (tobiasdiez) [inline .github/workflows/tests-oracle.yml]: > [warning]Cache size of ~2455 MB (2573776080 B) is over the 400MB limit, not saving cache.

I think github will not increase the cache size that dramatically in the near future. Thus, this cache can be removed (which reduces the build time by 2 mins - not much but better than nothing).

Comment (Siedlerchr) [reply .github/workflows/tests-oracle.yml]: For caching, see the discussion here: https://github.com/actions/cache/issues/6

Comment (koppor) [reply .github/workflows/tests-oracle.yml]: Result of the discussion:

> There may be another update soon to up the limit a little (~700MB or so), [...] We would still have the 2GB per repo limit until we integrate billing, so there would be some concern about thrashing the cache if the individual limit matched the per repo limit.

**Pull request merged**